crates/core/src/db/relational_db.rs

```diff
@@ -64,6 +69,7 @@ pub struct RelationalDB {
 
     inner: Locking,
     durability: Option<Arc<dyn Durability<TxData = Txdata>>>,
+    snapshot_worker: Option<Arc<SnapshotWorker>>,
```

Comment:
I wonder if it would be best to move this outside of the `RelationalDB` (`DatabaseEngine` future name), to something slightly higher level (e.g. `DatabaseInstanceContext`). That way we maintain that `RelationalDB` remains runtime independent. I don't feel it strongly enough to block the review, but it makes me uncomfortable to have the `DatabaseEngine`/`RelationalDB` dependent on tokio.

Reply:
To be clear, I think the ability to snapshot itself would remain inside of `RelationalDB`, I'm specifically just referring to the scheduling/triggering system.